bin/picca_fast_metal_dmat.py

```diff
@@ -0,0 +1,570 @@
+        stack_table_1: table
+            table with cols LOGLAM and WEIGHT for first series of deltas
+        stack_table_2: table
+            table with cols LOGLAM and WEIGHT for second series of deltas
+    Optionnal : rebin_factor
+            rebin loglam and weights
+    Returns:
+        The distortion matrix data
+        Note the global picca.cf contains the cosmology and the rp grid
+    """
+
+    loglam1=stack_table_1["LOGLAM"]
+    weight1=stack_table_1["WEIGHT"]
+    loglam2=stack_table_2["LOGLAM"]
+    weight2=stack_table_2["WEIGHT"]
+    if rebin_factor is not None :
+        n1=loglam1.size
+        loglam1 = loglam1[:(n1//rebin_factor)*rebin_factor].reshape((n1//rebin_factor),rebin_factor).mean(-1)
+        weight1 = weight1[:(n1//rebin_factor)*rebin_factor].reshape((n1//rebin_factor),rebin_factor).mean(-1)
+        n2=loglam2.size
+        loglam2 = loglam2[:(n2//rebin_factor)*rebin_factor].reshape((n2//rebin_factor),rebin_factor).mean(-1)
+        weight2 = weight2[:(n2//rebin_factor)*rebin_factor].reshape((n2//rebin_factor),rebin_factor).mean(-1)
+
+    # input
+    iz1 = (10**loglam1)/constants.ABSORBER_IGM[in_lambda_abs_1] - 1.
+    iz2 = (10**loglam2)/constants.ABSORBER_IGM[in_lambda_abs_2] - 1.
+    ir1 = cf.cosmo.get_r_comov(iz1)
+    ir2 = cf.cosmo.get_r_comov(iz2)
+    # all pairs
+    irp = (ir1[:,None]-ir2[None,:]).ravel() # same sign as line 676 of cf.py (1-2)
+    if not cf.x_correlation:
+        irp = np.abs(irp)
+    # output
+    oz1 = (10**loglam1)/constants.ABSORBER_IGM[out_lambda_abs_1] - 1.
+    oz2 = (10**loglam2)/constants.ABSORBER_IGM[out_lambda_abs_2] - 1.
+    or1 = cf.cosmo.get_r_comov(oz1)
+    or2 = cf.cosmo.get_r_comov(oz2)
+    # all pairs
+    orp = (or1[:,None]-or2[None,:]).ravel() # same sign as line 676 of cf.py (1-2)
+    if not cf.x_correlation:
+        orp = np.abs(orp)
+
+    # weights
+    sw  = ((weight1*((1+iz1)**(cf.alpha-1)))[:,None]*(weight2*((1+iz2)**(cf.alpha2-1)))[None,:]).ravel()
+
+    # distortion matrix
+    rpbins   = cf.r_par_min + (cf.r_par_max-cf.r_par_min)/cf.num_bins_r_par*np.arange(cf.num_bins_r_par+1)
+
+    # I checked the orientation of the matrix
+    dmat,_,_ = np.histogram2d(orp,irp,bins=(rpbins,rpbins),weights=sw)
+
+    # normalize (sum of weight should be one for each input rp,rt)
+    sum_in_weight,_ = np.histogram(irp,bins=rpbins,weights=sw)
+    dmat *= ((sum_in_weight>0)/(sum_in_weight+(sum_in_weight==0)))[None,:]
+
+    # mean outputs
+    sum_out_weight,_    = np.histogram(orp,bins=rpbins,weights=sw)
+    sum_out_weight_rp,_ = np.histogram(orp,bins=rpbins,weights=sw*(orp[None,:].ravel()))
+    sum_out_weight_z,_ = np.histogram(orp,bins=rpbins,weights=sw*(((oz1[:,None]+oz2[None,:])/2.).ravel()))
+    r_par_eff = sum_out_weight_rp/(sum_out_weight+(sum_out_weight==0))
+    z_eff     = sum_out_weight_z/(sum_out_weight+(sum_out_weight==0))
+
+    r_trans_eff = np.zeros(r_par_eff.shape)
+
+    if True :
```

Comment:
why the `if True`? why not simply continue with the normal bloc?

Reply:
done